test/integration/util/test_swagger_client.py

```diff
@@ -401,6 +405,57 @@ def test_put_with_invalid_enum_param(self):
                                              '--query-param', query_param_invalid])
             self.assertEqual(e.exception.code, 2)
 
+    def test_get_special_characters_in_path(self):
+        self._test("/with/special_characters.-/in/path", "get")
+
+    def test_get_with_header_parameter(self):
+        self._test("/with/header/parameter",
+                   "get",
+                   headers={"some_header": "foo"},
+                   command_args=["--some-header", "foo"])
+
+    def test_get_with_request_body_application_json(self):
+        self._test("/with/request_body/not_application/json",
+                   "get",
+                   json={"foo": "bar"},
+                   command_args=["--foo", "bar"])
+
+    def _test(self,
+              path: str,
+              http_method: str,
+              json: dict = None,
+              stream: bool = False,
+              params: dict = None,
+              headers: dict = None,
+              timeout = mock.ANY,
+              command_args: list = None):
+
+        params = params or {}
+        headers = headers or {}
+        command_args = command_args or []
+
+        command = self.get_command(http_method, path)
+        url = self.url_base + path
+
+        with mock.patch('requests.Session.request') as mock_request, \
+                mock.patch('hca.util._ClientMethodFactory._consume_response') as mock_consume_response:
+            mock_request.return_value = self.dummy_response
+
+            args = self.parser.parse_args([command] + command_args)
+            args.entry_point(args)
+            mock_consume_response.assert_called_once_with(mock.ANY)
+            mock_request.assert_called_once_with(http_method,
+                                                 url,
+                                                 json=json,
+                                                 stream=stream,
+                                                 params=params,
+                                                 headers=headers,
+                                                 timeout=timeout)
+
+    @staticmethod
+    def get_command(http_method, path):
+        return f"{http_method}{path}".replace('/', '-').replace('_', '-').replace('.', '')
```

Comment:
The operation that cleans the http path above is `http_path.replace('.', '').replace('-', '_')` but that's different from the one here. I think these two should be reconciled by putting this operation in a function in `hca/util/__init__.py` that can also be imported for the tests.

Reply:
Second this.